manual/src/cmds/tsan.etex

```diff
@@ -0,0 +1,227 @@
+conflicting access is made later by another thread. This does not constitute a
+data race, but TSan may report it as such.
+
+\section{s:tsan-runtime-flags}{Runtime options}
+
+TSan supports a number of configuration options at runtime using the
+\texttt{TSAN\_OPTIONS} environment variable. Notably, it allows to suppress
+some data races from TSan reports. See the
+\href{https://github.com/google/sanitizers/wiki/ThreadSanitizerFlags}{documentation
+of TSan flags} and the
+\href{https://github.com/google/sanitizers/wiki/SanitizerCommonFlags}{documentation
+of flags common to all sanitizers} for more information. Suppressing data race
+reports is useful for intentional races or libraries that cannot be fixed.
+
+For example, to suppress reports originating from functions in the OCaml module
+\texttt{My_module}, one could set
+\texttt{TSAN\_OPTIONS} to \texttt{"suppressions=suppr.txt"}, with
+\texttt{suppr.txt} a file containing:
+
+\begin{verbatim}
+race_top:^camlMy_module
+\end{verbatim}
+
+(Note that this depends on the format of OCaml symbols in the executable. Some
+builders, like Dune, might result in different formats. You should adapt this
+example to the symbols effectively present in your stack traces.)
+
+The \texttt{TSAN\_OPTIONS} variable also allows to increase the ``history
+size'', e.g.:
+
+\begin{verbatim}
+TSAN_OPTIONS="history_size=7" ./my_instrumented_program
+\end{verbatim}
+
+TSan’s history records events such as function entry and exit, and is used to
+reconstruct stack traces. Increasing the history size can sometimes be
+necessary to obtain the second stack trace, but it also increases memory
+consumption. This setting does not change the number of memory accesses
+remembered per memory location.
+
+\section{s:tsan-c-code}{Guidelines for linking}
+
+As a general rule, OCaml programs instrumented with TSan should only be linked
+with OCaml or C objects also instrumented with TSan. Doing otherwise can result
+in crashes. The only exception to this rule are C libraries that do not call
+into the OCaml runtime system in any way, i.e., do not allocate, raise
+exceptions, call back into OCaml code, etc. Examples include the libc or system
+libraries. Data races in non instrumented libraries will not be reported.
+
+C code interacting with OCaml should always be built through the
+\texttt{ocamlopt} command, which will pass the required instrumentation flags
+to the C compiler. The \texttt{CAMLno_tsan} qualifier can be used to prevent
+functions from being instrumented:
+
+\begin{verbatim}
+CAMLno_tsan void f(int arg)
+{
+  /* This function will not be instrumented. */
+  ...
+}
+\end{verbatim}
+
+Races from non instrumented functions will not be reported. Un-instrumenting
+functions should be done by expert users only. It can be used to reduce the
+performance overhead in certain corner cases, or to suppress some known alarms.
+For the latter, the mechanism of run-time suppressions should be preferred when
```

Comment:
What is run-time suppression and un-instrumentation? At this point, I don't know what they are or only have a vague idea. Is un-instrumentation `CAMLno_tsan` and runtime suppression is using "suppressions" text file? 


These terms were not described explicitly. If you need to use them, introduce them explicitly where they are first defined.

Reply:
I clarified by avoiding the term “un-instrument” and explicitly talking about either `TSAN_OPTIONS` or `CAMLno_tsan`.